### PR TITLE
[DNM]Crate chain-pushing

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: You can now push multiple crates in front of you. This also applies to conveyor belts.
/:cl:

If you're pushing a crate and there's a crate in front of that crate, you'll push that crate as well. Pretty much QoL for cargo, so they don't have to push every single crate one by one to get them on/off the shuttle.

### CODE REVIEW

**Comment by RemieRichards** [comment]
ruining sokoban

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 1]
```diff
@@ -19,6 +19,13 @@
 	..()
 	update_icon()
 
+/obj/structure/closet/crate/Bumped(AM as mob|obj)
+	..()
+	if(istype(AM, /obj/structure/closet/crate) && !anchored)
+		var/d = get_dir(AM, src)
+		if(step(src, d))
```
Comment (RemieRichards) [inline code/game/objects/structures/crates_lockers/crates.dm]: should this not be AM and the one below be src?

**Comment by Repukan** [comment]
Powercreep

File: code/game/objects/structures/crates_lockers/crates.dm (hunk 1 above)
Comment (XDTM) [inline code/game/objects/structures/crates_lockers/crates.dm]: AM is the pushing atom, if i do it the other way it'll just loop

Comment (RemieRichards) [inline code/game/objects/structures/crates_lockers/crates.dm]: Oh right, thought this was ```Bump()``` not ```Bumped()```

**Comment by Bawhoppen** [comment]
@RemieRichards I'll never understand that Sokoban thing

**Comment by RemieRichards** [comment]
push boxes

**Comment by Bawhoppen** [comment]
yea but like... why?

**Comment by Repukan** [comment]
It's fun

**Comment by RemieRichards** [comment]
>It's fun

**Comment by Hatfish** [comment]
This kills the boxfort.

**Comment by kevinz000** [comment]
wow since when were spacemen that buff?

**Comment by XDTM** [comment]
NT just secretly installed wheels on the crates

**Comment by optimumtact** [comment]
What's the limit on the chain?

**Comment by XDTM** [comment]
currently none, is it abusable?

**Comment by Jalleo** [comment]
Set up a circle all but one tile conveyer crates on all the conveyer. Will it infinite loop?

**Comment by optimumtact** [comment]
Just for sanities sake there should be a limit of 3-4 crates.

**Comment by Anonmare** [comment]
Yeah put a hard limit on it. 4-5 at max

**Comment by Dorsisdwarf** [comment]
I dont like this, like 90% of cargo is wrangling crates so that you never end up pushing more than one at once

**Comment by coiax** [comment]
We could move Cargo into more about actually ordering and delivering materials that the station needs

**Comment by Dorsisdwarf** [comment]
but people can either get them already in a different way or dont need them in 90% of rounds

**Comment by XDTM** [comment]
Pretty much, my long term plan was to make cargo less clunky to work on while giving it a more defined niche and maybe progression.

**Comment by Iamgoofball** [comment]
@Dorsisdwarf if a job is defined by a clunky annoying feature its time to replace it

**Comment by XDTM** [comment]
Current code crashes the MC when the crates are on a conveyor belt.
I'm leaving the PR open for a bit if someone has a better idea for the implementation, because i can't find a decent way to code this.

**Comment by BeeSting12** [comment]
honestly im not a huge fan of this in the first place since it'll mess up crates blocking paths and stuff which is funny

**Comment by ChangelingRain** [comment]
make it only happen if the original bumper is a living mob who can reasonably interact with the crates.

**Pull request closed without merging**